src/transaction.ts

```diff
@@ -139,9 +139,7 @@ export function parseRawSimulation(
     minResourceFee: sim.minResourceFee,
     latestLedger: sim.latestLedger,
     cost: sim.cost,
-    transactionData: new SorobanDataBuilder(
-      sim.transactionData === "" ? undefined : sim.transactionData
-    ),
+    transactionData: new SorobanDataBuilder(sim.transactionData),
```

Comment:
@willemneal just fyi, this was fixed by https://github.com/stellar/js-stellar-base/pull/686